sycl/test-e2e/bindless_images/mipmap/mipmap_read_3D.cpp

```diff
@@ -133,10 +143,44 @@ int main() {
     }
   }
   if (validated) {
-    std::cout << "Test passed!" << std::endl;
-    return 0;
+    return false;
+  }
+
+  return true;
+}
+
+int main() {
+
+  bool failed = false;
+
+  failed |=
+      runTest<int, sycl::image_channel_type::signed_int32, class intTest>();
+
+  failed |=
+      runTest<uint, sycl::image_channel_type::unsigned_int32, class uintTest>();
+
+  failed |= runTest<float, sycl::image_channel_type::fp32, class floatTest>();
+
+  failed |=
+      runTest<short, sycl::image_channel_type::signed_int16, class shortTest>();
+
+  failed |= runTest<ushort, sycl::image_channel_type::unsigned_int16,
+                    class ushortTest>();
+
+  failed |=
+      runTest<char, sycl::image_channel_type::signed_int8, class charTest>();
+
+  failed |= runTest<unsigned char, sycl::image_channel_type::unsigned_int8,
+                    class ucharTest>();
+
+  failed |=
+      runTest<sycl::half, sycl::image_channel_type::fp16, class halfTest>();
```

Comment:
Personally I like to count the number of failing test and return that in main. It makes the end of the file a little simpler and it makes the return value mean something more than just some magical 1 (or 3 in this case.)

Reply:
Really good suggestion. Updated to use this. Will likely use this pattern now on.